forc/src/utils/cli_error.rs

```diff
@@ -1,91 +1,67 @@
-use std::net::AddrParseError;
+use anyhow::{Error, Result};
 use std::path::PathBuf;
-use std::{fmt, io};
 
-use sway_core::CompileError;
+use forc_pkg::Manifest;
+use forc_util::find_manifest_dir;
+use sway_core::{parse, CompileError};
 use sway_utils::constants::MANIFEST_FILE_NAME;
 
-#[derive(Debug)]
-pub struct CliError {
-    pub message: String,
+pub fn manifest_file_missing(curr_dir: PathBuf) -> anyhow::Error {
+    let message = format!(
+        "could not find `{}` in `{}` or any parent directory",
+        MANIFEST_FILE_NAME,
+        curr_dir.display()
+    );
+    Error::msg(message)
 }
 
-impl CliError {
-    pub fn manifest_file_missing(curr_dir: PathBuf) -> Self {
-        let message = format!(
-            "could not find `{}` in `{}` or any parent directory",
-            MANIFEST_FILE_NAME,
-            curr_dir.display()
-        );
-        Self { message }
-    }
-
-    pub fn parsing_failed(project_name: &str, errors: Vec<CompileError>) -> Self {
-        let message = errors
-            .iter()
-            .map(|e| e.to_friendly_error_string())
-            .collect::<Vec<String>>()
-            .join("\n");
-
-        Self {
-            message: format!("Parsing {} failed: \n{}", project_name, message),
-        }
-    }
-
-    pub fn wrong_sway_type(project_name: &str, wanted_type: &str, parse_type: &str) -> Self {
-        let message = format!(
-            "{} is not a '{}' it is a '{}'",
-            project_name, wanted_type, parse_type
-        );
-        Self { message }
-    }
-
-    pub fn fuel_core_not_running(node_url: &str) -> Self {
-        let message = format!("could not get a response from node at the URL {}. Start a node with `fuel-core`. See https://github.com/FuelLabs/fuel-core#running for more information", node_url);
-        Self { message }
-    }
-}
-
-impl fmt::Display for CliError {
-    fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
-        write!(f, "{}", self)
-    }
+pub fn parsing_failed(project_name: &str, errors: Vec<CompileError>) -> anyhow::Error {
+    let error = errors
+        .iter()
+        .map(|e| e.to_friendly_error_string())
+        .collect::<Vec<String>>()
+        .join("\n");
+    let message = format!("Parsing {} failed: \n{}", project_name, error);
+    Error::msg(message)
 }
 
-impl From<&str> for CliError {
-    fn from(s: &str) -> Self {
-        CliError {
-            message: s.to_string(),
-        }
-    }
+pub fn wrong_sway_type(project_name: &str, wanted_type: &str, parse_type: &str) -> anyhow::Error {
```

Comment:
```suggestion
pub fn wrong_project_type(project_name: &str, wanted_type: &str, parse_type: &str) -> anyhow::Error {
```
Or
```suggestion
pub fn wrong_project_type(project_name: &str, wanted_type: &str, parse_type: &str) -> anyhow::Error {
```
I think those names would both be more descriptive.

Reply:
You've suggested `wrong_project_type` twice? :slightly_smiling_face:

Reply:
I agree with both versions 🙂

Reply:
Oh I meant wrong tree type or wrong project type. My bad!